### PR TITLE
Np 46910 brage merging should lookup several places

### PR DESCRIPTION
**Pull request by anetteOlli**
Tiny problem, gigantic refactor.

### CODE REVIEW

File: brage-import/src/main/java/no/sikt/nva/brage/migration/merger/findExistingPublicationService/CristinIdentifierFinder.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import no.sikt.nva.brage.migration.lambda.MergeSource;
+import no.sikt.nva.brage.migration.lambda.PublicationComparator;
+import no.sikt.nva.brage.migration.merger.DuplicatePublicationException;
+import no.sikt.nva.brage.migration.model.PublicationForUpdate;
+import no.sikt.nva.brage.migration.model.PublicationRepresentation;
+import no.unit.nva.model.AdditionalIdentifier;
+import no.unit.nva.model.Publication;
+import no.unit.nva.publication.service.impl.ResourceService;
+
+public class CristinIdentifierFinder implements FindExistingPublicationService {
```
Comment (anetteOlli) [inline brage-import/src/main/java/no/sikt/nva/brage/migration/merger/findExistingPublicationService/CristinIdentifierFinder.java:16]: Isolated each way we search for duplicates in their own methods.

File: brage-import/src/main/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumer.java [hunk 2]
```diff
@@ -71,22 +64,11 @@ public class BrageEntryEventConsumer implements RequestHandler<S3Event, Publicat
...
-    public static final String TITLE = "title";
-    public static final String CONTEXT_TYPE = "contextType";
-    public static final String AGGREGATION = "aggregation";
-    public static final String NONE = "none";
-    public static final String ISBN = "isbn";
     private final S3Client s3Client;
     private final ResourceService resourceService;
-    private String brageRecordFile;
-    private List<Publication> publicationsToMerge;
-    private MergeSource source;
```
Comment (anetteOlli) [inline brage-import/src/main/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumer.java:87]: Finally got rid of these two variables. I know why they were there, but I want them gone.

A function should either return a result OR modify environment, never both.

File: brage-import/src/main/java/no/sikt/nva/brage/migration/model/PublicationForUpdate.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package no.sikt.nva.brage.migration.model;
+
+import no.sikt.nva.brage.migration.lambda.MergeSource;
+import no.unit.nva.model.Publication;
+
+public record PublicationForUpdate(MergeSource source, Publication existingPublication) {
```
Comment (anetteOlli) [inline brage-import/src/main/java/no/sikt/nva/brage/migration/model/PublicationForUpdate.java:6]: The two variables are now stored here.

File: brage-import/src/main/java/no/sikt/nva/brage/migration/merger/findExistingPublicationService/FindExistingPublicationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            () -> cristinIdentifierFinder.findExistingPublication(publicationRepresentation),
+            () -> doiFinder.findExistingPublication(publicationRepresentation),
+            () -> isbnFinder.findExistingPublication(publicationRepresentation),
+            () -> titleAndTypeFinder.findExistingPublication(publicationRepresentation));
+
+        return updatePublicationSuppliers.stream()
+                   .map(Supplier::get)
+                   .filter(Optional::isPresent)
+                   .findFirst()
+                   .orElse(Optional.empty());
```
Comment (anetteOlli) [inline brage-import/src/main/java/no/sikt/nva/brage/migration/merger/findExistingPublicationService/FindExistingPublicationServiceImpl.java:40]: The idea of these are to stop searching for duplicates as soon as we find a single hit.

**Review by truhacevkir — APPROVED** [review]

**Pull request merged**